File: move_zeros.cxx
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        auto j = nums.begin();
        for(auto i : nums) {
            if(!i) continue;
            *j++ = i;
        }
        std::fill(j,nums.end(),0);
    }
};

File: move_zeros.py
class Solution:
    #def moveZeroes(self, nums: List[int]) -> None:
    #    j=0
    #    for i in nums:
    #        if i:
    #            nums[j] = i
    #            j+=1
    #    for i in range(j,len(nums)):
    #        nums[i]=0

    #def moveZeroes(self, nums: List[int]) -> None:
    #    j=0
    #    for i in range(len(nums)):
    #        if nums[i]:
    #            nums[i],nums[j] = nums[j],nums[i]
    #            j+=1

    def moveZeroes(self, nums: List[int]) -> None:
        for i in nums:
            if i==0:
                nums.append(0)
                nums.remove(0)
